gcp-gke/main.tf

```diff
@@ -0,0 +1,45 @@
+
+module "gke" {
+  source  = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
+  version = "~> 29.0"
+
+  project_id                        = var.project_id
+  name                              = var.cluster_name
+  region                            = var.region
+  regional                          = false
+  zones                             = var.zones
+  network                           = var.network_name
+  subnetwork                        = var.subnetwork_name
+
+  //dedicated Secondary IP range for pods
+  ip_range_pods                     = "pod-ip-range"
+  //dedicated Secondary IP range for services
+  ip_range_services                 = "service-ip-range"
+  
+  enable_private_endpoint           = true
+  enable_private_nodes              = true
+  master_ipv4_cidr_block            = "172.16.0.0/28"
+  network_policy                    = true
+  horizontal_pod_autoscaling        = true
+  service_account                   = "create"
+  remove_default_node_pool          = true
+  disable_legacy_metadata_endpoints = true
+  deletion_protection               = var.deletion_protection
+  deploy_using_private_endpoint     = true
+  enable_shielded_nodes             = true
+  enable_vertical_pod_autoscaling   = var.enable_vertical_pod_autoscaling
+  config_connector                  = var.config_connector
+  cluster_autoscaling               = var.cluster_autoscaling
+  master_authorized_networks = [
```

Comment:
There are some attributes that I would like to be able to set through variables. We can still set sensible defaults for these in the `variables.tf` file. I'm thinking mostly of differences between clusters in dev / stg / prod, so we can reduce cost on non-crucial clusters:
- `regional`: we want regional in prod and zonal everywhere else
- `enable_private_nodes`, `enable_shielded_nodes`: don't know if we incur in extra cost by setting these, or even added complexity which brings no benefit in dev / stg
- `master_ipv4_cidr_block`: might be interesting to expose this one as in the future we might want to have tailscale subnet router operators, and need different cidr block on all clusters